### PR TITLE
Tbolt/2406 update save error to modal

### LINKED ISSUE #2406: Update "unable to save" warning to use modal pattern
Update the "unexpected error saving" message to use the modal pattern provided by design.cms.gov

Update the text on the save changes error pop-up

Header: Unable to save changes!

Body: Your changes aren't being saved. Try saving your changes again in a few minutes by clicking on the save button. If you refresh your browser without saving backing up your changes, changes made after this message appeared will be lost.

### This task is done when...
 - [x] Old message pattern is removed
 - [x] "Unexpected error saving" now uses the modal pattern
 - [x] Update the text in the modal to the above header/body text

### PR DESCRIPTION
**Pull request by tbolt**
Resolves #2406 and #2803 

### This pull request changes...

- Updates "Unable to save" error to use a modal
- Updates the text in the "Unable to save" modal

### This pull request is ready to merge when...

- [ ] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- [x] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [ ] The change has been documented
  - [ ] Associated OpenAPI documentation has been updated
- [x] Changelog is updated as appropriate

### This feature is done when...

- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

**Review by thetif — APPROVED** [review]

File: web/src/components/Header.test.js [hunk 1]
```diff
@@ -103,7 +103,7 @@ describe('Header component', () => {
       role: 'eAPD State Admin',
       state: { id: 'wa', name: 'Washington' },
       username: 'frasiercrane@kacl.com'
-    }
+    };
```
Comment (eStar386) [inline web/src/components/Header.test.js:106]: I love the coding format, it's prettier.

**Comment by eStar386** [comment]
Exemption Documented: Testing showed accessibility issues on various user interfaces and/or tools. This is isn't a fault of the added code but are possibly known issues. That need to be addressed before closing the issue.

**Review by eStar386 — APPROVED** [review]

**Comment by jeromeleecms** [comment]
@beparticular @tbolt 

"Your changes aren't being saved. Try saving your changes again in a few minutes by clicking on the save button. If you refresh your browser without **saving backing up** your changes, changes made after this message appeared will be lost."

Are we missing a word between saving and backing? Alternatively, since you can't back up changes since the modal won't close and you can't access content underneath unless a save is completed:

"Your changes aren't being saved. **Please verify your internet connection and** try saving your changes again in a few minutes by clicking on the save button. If you refresh your browser without saving, changes made after this message appeared will be lost."

Also, what happens if the network connection is never restored? The modal never goes away (and you can't escape it) - what are we expecting the user to do after this if they can't save? Now that I'm testing/reading, we aren't giving much guidance as to what they might try to do before contacting the helpdesk.

Thoughts?

**Comment by beparticular** [comment]
@jeromeleecms 

1. Let's just remove "backing up your changes" in that paragraph. (I feel like a part of this discussion disappeared somewhere...) 

2. Is a broken internet connection the only reason this error could show? @thetif

**Comment by thetif** [comment]
It could probably also happen if the server went down.

**Comment by beparticular** [comment]
"Your changes aren't being saved. Verify your internet connection and try saving your changes again in a few minutes by clicking on the save button. If you refresh your browser without saving, changes made after this message appeared will be lost.

If your internet connection is working and this message persists for more than 5 minutes, please contact us at CMS-EAPD@cms.hhs.gov."

**Comment by tbolt** [comment]
@jeromeleecms @beparticular Copy has been updated.

**Comment by beparticular** [comment]
@tbolt there's an interesting thing happening in the background on this modal. 

If the autosave message was in the state where it just says "Saved!" when I lose my connection and I hit the "Save" button on the modal, I'm seeing the autosave message look like it's updating - the spinner twirls and says "Saved!" in the background even when I'm offline.  This seems a little confusing. 

If the state of the autosave message had minutes since last saved on it before I lost my connection and hit the save button on the modal, it's also updating the number of minutes it's been since I last saved. This seems like the right behavior. 

Is there anything we should do about this?

**Comment by tbolt** [comment]
@beparticular I made a ticket for that https://github.com/CMSgov/eAPD/issues/2830. I agree though, it shouldn't say Saved! when it's failing.  

@jeromeleecms I found a fix for the close icon not loading. Test it out and let me know if you are still seeing the issue.

**Review by beparticular — APPROVED** [review]

**Review by jeromeleecms — APPROVED** [review]
Overall looks good - I noticed that on initial click, the modal still tries to access the online icon, but eventually it "sticks." I don't think it's worth spending a ton more time to try and resolve and it doesn't affect system functionality.

**Comment by akuas** [comment]
@beparticular I updated the copy in the modal design.

**Pull request merged**